pkg/manifest/manifest.go

```diff
@@ -203,35 +214,40 @@ func (m *Manifest) Include(excludeIdentifier *string, requiredFeatureSet *string
 
 	// If there is no capabilities defined in a release then we do not need to check presence of capabilities in the manifest
 	if capabilities != nil {
-		return checkResourceEnablement(annotations, capabilities)
+		return checkResourceEnablement(annotations, capabilities, allowUnknownCapabilities)
 	}
 	return nil
 }
 
 // checkResourceEnablement, given resource annotations and defined cluster capabilities, checks if the capability
-// annotation exists. If so, each capability name is validated against the known set of capabilities. Each valid
-// capability is then checked if it is disabled. If any invalid capabilities are found an error is returned listing
-// all invalid capabilities. Otherwise, if any disabled capabilities are found an error is returned listing all
-// disabled capabilities.
-func checkResourceEnablement(annotations map[string]string, capabilities *configv1.ClusterVersionCapabilitiesStatus) error {
+// annotation exists. If so, each capability name is validated against the known set of capabilities unless
+// allowUnknownCapabilities is true. Each valid capability is then checked if it is disabled. If any invalid
+// capabilities are found an error is returned listing all invalid capabilities. Otherwise, if any disabled
+// capabilities are found an error is returned listing all disabled capabilities.
+func checkResourceEnablement(annotations map[string]string, capabilities *configv1.ClusterVersionCapabilitiesStatus,
+	allowUnknownCapabilities bool) error {
+
 	caps := getManifestCapabilities(annotations)
 	numCaps := len(caps)
 	unknownCaps := make([]string, 0, numCaps)
 	disabledCaps := make([]string, 0, numCaps)
 
 	for _, c := range caps {
-		var isKnownCap bool
-		var isEnabledCap bool
 
-		for _, knownCapability := range capabilities.KnownCapabilities {
-			if c == knownCapability {
-				isKnownCap = true
+		if !allowUnknownCapabilities {
```

Comment:
I'm fine as you have it, but a chattier option would be to leave this `unknownCaps` population in place, and later on do something like:

```go
if len(unknownCaps) > 0 {
	if allowUnknownCapabilities {
		klog.V(2).Infof("allowing unrecognized capability names: %s", strings.Join(unknownCaps, ", "))
	} else {
		return fmt.Errorf("unrecognized capability names: %s", strings.Join(unknownCaps, ", "))
	}
}
```

Although maybe to make that more usable, you'd need to make this helper a method:

```go
func (m *Manifest) checkResourceEnablement(...)
```

so you could format `m` into the log line?

Reply:
Prefer to leave it as is